src/popup.js

```diff
@@ -119,49 +123,52 @@ function closeOverlay() {
  */
 function send_error(message) {
   var browser = window.navigator.userAgent;
-  var tabId = parseInt($('#associatedTab').attr('data-tab-id'), 10);
-  var origins = backgroundPage.getAllOriginsForTab(tabId);
-  if(!origins){ return; }
-  var version = chrome.runtime.getManifest().version;
-  //TODO "there's got to be a better way!"
-  var fqdn = tab.url.split("/",3)[2];
-  var out = {"browser":browser, "url":tab.url,"fqdn":fqdn, "message":message, "version": version};
-  for (var i = 0; i < origins.length; i++){
-     var origin = origins[i];
-     var action = backgroundPage.getAction(tabId, origin);
-     if (!action){ action = pb.NO_TRACKING; }
-     if (out[action]){
-       out[action] += ","+origin;
-     }
-     else{
-       out[action] = origin;
-     }
-  }
-  var out_data = JSON.stringify(out);
-  console.log(out_data);
-  var sendReport = $.ajax({
-    type: "POST",
-    url: "https://privacybadger.org/reporting",
-    data: out_data,
-    contentType: "application/json"
-  });
-  sendReport.done(function() {
-    $("#error_input").val("");
-    $("#report_success").toggleClass("hidden", false);
-    setTimeout(function(){
-      $("#report_button").prop("disabled", false);
-      $("#report_cancel").prop("disabled", false);
-      $("#report_success").toggleClass("hidden", true);
-      closeOverlay();
-   }, 3000);
-  });
-  sendReport.fail(function() {
-    $("#report_fail").toggleClass("hidden");
-    setTimeout(function(){
-      $("#report_button").prop("disabled", false);
-      $("#report_cancel").prop("disabled", false);
-      $("#report_fail").toggleClass("hidden", true);
-   }, 3000);
+  getTab(function(tab) {
+    var tabId = tab.id;
+    var badger = backgroundPage.getBadgerWithTab(tabId);
+    var origins = badger.getAllOriginsForTab(tabId);
+    if(!origins){ return; }
+    var version = chrome.runtime.getManifest().version;
+    //TODO "there's got to be a better way!"
+    var fqdn = tab.url.split("/",3)[2];
+    var out = {"browser":browser, "url":tab.url,"fqdn":fqdn, "message":message, "version": version};
+    for (var i = 0; i < origins.length; i++){
+       var origin = origins[i];
+       var action = backgroundPage.getAction(tabId, origin);
+       if (!action){ action = constants.NO_TRACKING; }
+       if (out[action]){
+         out[action] += ","+origin;
+       }
+       else{
+         out[action] = origin;
+       }
+    }
+    var out_data = JSON.stringify(out);
+    console.log(out_data);
```

Comment:
This is present in *master* but should we use `backgroundPage.log(..)` instead of all the `console.log(...)` calls in popup.js?